doc/hooks/versionCheckHook.section.md

```diff
@@ -22,14 +24,32 @@ stdenv.mkDerivation (finalAttrs: {
 })
 ```
 
-Note that for [`buildPythonPackage`](#buildpythonpackage-function) and [`buildPythonApplication`](#buildpythonapplication-function), `doInstallCheck` is enabled by default.
+Note that some builders, e.g. [`buildPythonPackage`](#buildpythonpackage-function) and [`buildPythonApplication`](#buildpythonapplication-function), enable `doInstallCheck` by default.
 
 It does so in a clean environment (using `env --ignore-environment`), and it checks for the `${version}` string in both the `stdout` and the `stderr` of the command. It will report to you in the build log the output it received and it will fail the build if it failed to find `${version}`.
 
-The variables that this phase control are:
+This hook is controlled by the following attributes:
+
+- `dontVersionCheck`: When set as `true`, the hook is not added to the [`preInstallCheckHooks`](#ssec-installCheck-phase).
+
+  Useful when one wants to load the function defined by the hook, so that it can be used for a different purpose.
+
+  Defaults to false.
+
+- `versionCheckProgram`: The full path to the program to be executed.
+
+   Defaults to `${placeholder "out"}/bin/${pname}`, roughly speaking.
+
+   The usage of `placeholder "out"` instead of `$out` here is unavoidable, since the hook does not expand environment variables.
+
+- `versionCheckProgramArg`: The argument to be passed to `versionCheckProgram`.
+
+  When not defined, the hook tries `--help` first and then `--version`.
+
+  Examples: `version`, `-V`, `-v`.
+
+- `preVersionCheck`: Hook that runs before the check is done. Defaults to an empty string.
+
+- `postVersionCheck`: Hook that runs after the check is done. Defaults to an empty string.
 
-- `dontVersionCheck`: Disable adding this hook to the [`preInstallCheckHooks`](#ssec-installCheck-phase). Useful if you do want to load the bash functions of the hook, but run them differently.
-- `versionCheckProgram`: The full path to the program that should print the `${version}` string. Defaults roughly to `${placeholder "out"}/bin/${pname}`. Using `$out` in the value of this variable won't work, as environment variables from this variable are not expanded by the hook. Hence using `placeholder` is unavoidable.
-- `versionCheckProgramArg`: The argument that needs to be passed to `versionCheckProgram`. If undefined the hook tries first `--help` and then `--version`. Examples: `version`, `-V`, `-v`.
-- `preVersionCheck`: A hook to run before the check is done.
-- `postVersionCheck`: A hook to run after the check is done.
+Note: see also [`testers.testVersion`](#tester-testVersion).
```

Comment:
Once again, it is a bit annoying IMO to read this and to see that there is a completely different alternative and not having any reference to a comparison between the two.

Reply:
I moved the comparison to this file.